File: Actividad5/Ejercicio1/CarbonFootprintProject/src/model/Building.java
package model;

public class Building implements CarbonFootprint {

    private double electricityUsage; // kWh

    public Building(double electricityUsage) {
        this.electricityUsage = electricityUsage;
    }

    @Override
    public double getCarbonFootprint() {

        if (electricityUsage < 0) {
            return 0;
        }

        return electricityUsage * 0.5; // factor estimado
    }

    @Override
    public String toString() {
        return "Building | Electricity: " + electricityUsage;
    }
}

File: Actividad6/Ejercicio 1/CarbonFootprintPy/model/building.py
from model.CarbonFootprint import CarbonFootprint
from model.exceptions import InvalidEmissionDataError

class Building(CarbonFootprint):
    def __init__(self, electricity_usage: float):
        if electricity_usage < 0:
            raise InvalidEmissionDataError(f"Uso de electricidad inválido: {electricity_usage}. No puede ser negativo.")
        self.electricity_usage = electricity_usage

    def get_carbon_footprint(self) -> float:
        return self.electricity_usage * 0.5  # factor estimado

    def __str__(self):
        return f"Building | Electricity: {self.electricity_usage} kWh"